sys/parsoid_proxy.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+/*
+ * Simple proxy to route requests to the client-requested
+ * Parsoid variant (JS or PHP) during the transition
+ * period. Parsoid/JS is being phased out and replaced by
+ * Parsoid/PHP.
+ */
+
+const HyperSwitch = require('hyperswitch');
+const URI = HyperSwitch.URI;
+const mwUtil = require('../lib/mwUtil');
+const VARIANT_HDR_NAME = 'x-parsoid-variant';
+
+module.exports = () => {
+    return {
+        spec: {
+            paths: {
+                '/{+path}': {
+                    all: {
+                        operationId: 'proxy_parsoid_variant'
+                    }
+                }
+            }
+        },
+        operations: {
+            proxy_parsoid_variant: (hyper, req) => {
+                const rootReqHeaders = hyper._rootReq.headers || {};
+                rootReqHeaders[VARIANT_HDR_NAME] = rootReqHeaders[VARIANT_HDR_NAME] || 'JS';
+                const isPhpVariant = /PHP/i.test(rootReqHeaders[VARIANT_HDR_NAME]);
+
+                return hyper.request({
+                    method: req.method,
+                    uri: new URI(req.uri.toString().replace('/parsoid/',
+                        isPhpVariant ? '/parsoidphp/' : '/parsoidjs/')),
+                    headers: req.headers,
+                    body: req.body,
+                    query: req.query,
+                    params: req.params
+                })
+                .then((res) => {
+                    if (res) {
+                        res.headers = res.headers || {};
+                        res.headers[VARIANT_HDR_NAME] = rootReqHeaders[VARIANT_HDR_NAME];
```

Comment:
I'm not quite sure this is a right behavior that wouldn't accidentally mask issues. We `assume` we've routed the request to correct Parsoid, but who knows? Why would we set the response header for parsed variant here??

Reply:
It's in the ticket: "Furthermore, RESTBase must include this header in the response, ..."  https://phabricator.wikimedia.org/T230791

Reply:
Ideally, the variant header would be added by the Parsoid instances instead of Restbase

Reply:
Yeah, that's what I meant. Or, the content-type would be different. Would you mind filing a ticket, something like 'Parsoid JS/PHP should indicate which Parsoid implementation the page was rendered with'?